staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -196,7 +207,11 @@ func (wc *watchChan) startWatching(watchClosedCh chan struct{}) {
 			return
 		}
 	}
-	opts := []clientv3.OpOption{clientv3.WithRev(wc.initialRev + 1), clientv3.WithPrevKV()}
+	opts := []clientv3.OpOption{
+		clientv3.WithRev(wc.initialRev + 1),
+		clientv3.WithPrevKV(),
+		clientv3.WithProgressNotify(),
```

Comment:
Quick question now: what will happen if I call it against etcd in a version that doesn't support it (e.g. 3.0.*)?

Reply:
etcd-3.0 supports this feature already, it will be omitted even if server does not support it.

Reply:
Ahh - I though this is new feature.
So one more question I have is - how frequently those "progress notify" events are being send?

I'm trying to understand the impact of it and the gain we get from just this PR.

Reply:
etcd server sends progress notify every 10 minutes, which is too long for kubernetes apiserver. With https://github.com/etcd-io/etcd/pull/9869/files, we should request progress notify once 1 minute.
Send a progress notify request to etcd 3.3 will just be omitted, which means we can do that safely without an 'etcd version detector'.

Reply:
Hmm - so I don't really understand what this PR is supposed to give us. I don't see any real benefit from it...